### PR TITLE
Android O updates

### PR DESCRIPTION
**Pull request by secondsun**
This requires an updated version of Aerogear Android Push and can be found here : https://github.com/aerogear/aerogear-android-push/pull/74

### CODE REVIEW

File: app/src/main/java/org/feedhenry/pushstarter/handler/NotificationBarMessageHandler.java [hunk 1]
```diff
@@ -69,6 +78,35 @@ public void fail(FHResponse fhResponse) {
...
 
+    @RequiresApi(api = Build.VERSION_CODES.O)
+    private void createChannel() {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+            throw new IllegalStateException("This function should not be called on < Android O");
+        }
+
+        NotificationManager notificationManager =
+                (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
+        // The id of the channel.
```
Comment (damienomurchu) [inline app/src/main/java/org/feedhenry/pushstarter/handler/NotificationBarMessageHandler.java]: redundant comment line

**Review by damienomurchu — COMMENTED** [review]
Code looks fine, just need to update AeroGear Android Push to test fully before approving

File: build.gradle [hunk 2]
```diff
@@ -16,5 +16,9 @@ buildscript {
 allprojects {
     repositories {
         jcenter()
+        mavenLocal()
```
Comment (damienomurchu) [inline build.gradle:20]: leave in the final version?

**Review by damienomurchu — APPROVED** [review]
Push notifications received properly when app both in foreground and background. Notification badge displaying correctly on app icon when push notifications received and app is not in focus.

Code looks good to me, aside from the one/ two small comments. 

Steps taken to verify:
- code inspected
- pr changes & aerogear-android-push changes pulled & built
- built apps run in api 25 & O emulator images, connected to push cloud app on devexp cluster, and notifications sent to app

Assuming CI build is just failing as aerogear-android-push pr change which this pr relies on, hasn't been merged yet

File: app/build.gradle [hunk 3]
```diff
@@ -5,25 +5,17 @@ plugins {
...
 
 android {
-    compileSdkVersion 25
-    buildToolsVersion '25.0.3'
+    compileSdkVersion 26
+    buildToolsVersion '26.0.1'
 
     defaultConfig {
-        applicationId "org.feedhenry.pushstarter"
+        applicationId "org.feedhenry.welcome"
```
Comment (damienomurchu) [inline app/build.gradle]: any reason for this change by the way?

File: build.gradle (hunk 2 above)
Comment (secondsun) [reply build.gradle:20]: I would say remove it after we merge and release the AeroGear library.

File: app/build.gradle (hunk 3 above)
Comment (secondsun) [reply app/build.gradle]: I am sure it was a deep and profound copy paste error ;)

**Comment by danielpassos** [comment]
@secondsun Can you also update the circle-ci? :kissing_heart:

**Review by danielpassos — APPROVED** [review]

**Comment by danielpassos** [comment]
circle-ci is complaining because we didn't release the AeroGear Android Push 4.2.0 since it's going to Android-O branch there is no problem. 

Landed at Android-O branch https://github.com/feedhenry-templates/pushstarter-android-app/commit/1d728a32153f12bfece10d4bf00c8c899526fe18

**Pull request closed without merging**